pkg/network/kube_proxy.go

```diff
@@ -121,7 +121,12 @@ func FillKubeProxyDefaults(conf, previous *operv1.NetworkSpec) {
 	}
 
 	if conf.KubeProxyConfig.BindAddress == "" {
-		conf.KubeProxyConfig.BindAddress = "0.0.0.0"
+		// TODO: this will probably need to change for dual stack
+		if net.ParseIP(conf.ClusterNetwork[0].CIDR).To4() != nil {
```

Comment:
`ParseIP` takes an IP address not a CIDR string so this won't work right.

There are some nicer "is this IPv6" methods in upstream k8s now. Add `utilnet "k8s.io/utils/net"` to the imports, and use `utilnet.IsIPv6CIDRString(conf.ClusterNetwork[0].CIDR)` here. (Well, and put a "`!`" before it, or swap the then and else clauses around.)

Reply:
I went to replace `ParseIP` with `ParseCIDR` but the k8s method is much better. Changed this around.